src/pages/index.js

```diff
@@ -25,26 +36,109 @@ const cardData = {
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    console.log("Setting avatar to:", userData.avatar);
+    console.log("User Data from API:", userData);
+
+    userInfo.setUserInfo({
+      name: userData.name,
+      job: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((error) => {
+    console.error("Error setting userInfo:", error);
+  });
+
+/* ------------------------------------------------- */
+/*                     Form Validation
+/* ------------------------------------------------- */
+const formValidators = {};
+
+const enableValidation = (settings) => {
+  const formList = Array.from(document.querySelectorAll(settings.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(settings, formElement);
+    const formName = formElement.getAttribute("name");
+
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(settings);
+
 // /* ------------------------------------------------- */
-// /*                 Profile Edit Form
+// /*                 Profile Avatar
 /* ------------------------------------------------- */
+const handleAvatarFormSubmit = (formData) => {
+  const newAvatarUrl = formData.avatarUrl;
 
+  return api.updateAvatar(newAvatarUrl).then((userData) => {
+    console.log("Avatar updated on server:", userData.avatar);
+    userInfo.setUserAvatar(userData.avatar);
+  });
+};
+
+const profileEditAvatar = new PopupWithForm(
+  "#update-avatar-modal",
+  handleAvatarFormSubmit,
+  (formValues) => {
+    profileEditAvatar.renderLoading(true);
+    const avatarUrl = formValues.avatar;
+    api
+      .updateAvatar(avatarUrl)
+      .then((userData) => {
+        console.log("Avatar URL:", userData.avatar);
+        userInfo.setUserAvatar(userData.avatar);
+        profileEditAvatar.close();
+      })
+      .catch((err) => {
+        console.error(`Error Submitting Form: ${err}`);
+      });
+  }
+);
+
+const profileEditAvatarButton = document.querySelector(".profile__edit-icon");
+
+profileEditAvatar.setEventListeners();
+
+profileEditAvatarButton.addEventListener("click", () => {
+  profileEditAvatar.open();
+});
+
+// /* ------------------------------------------------- */
+// /*                 Profile Edit
+/* ------------------------------------------------- */
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   (formValues) => {
-    userInfo.setUserInfo({
-      name: formValues.title,
-      job: formValues.description,
-    });
+    profileEditPopup.renderLoading(true);
```

Comment:
Please, delete it

You have everything inside `PopupWithForm`. You start the loading effect, close the modal, catch errors, return the button text back in `finally`. So you don't need to repeat it in `index.js` again

**Please, correct it everywhere in the code**